remotespark/sparkkernelbase.py

```diff
@@ -134,9 +137,10 @@ def _register_auto_viz(self):
                            log_if_error="Failed to register auto viz for notebook.")
         self.logger.debug("Registered auto viz.")
 
-    def _start_session(self):
+    def _start_session(self, msg_printer):
```

Comment:
`msg_printer` shouldn't be an argument to all these methods -- it should be an instance variable.  

i.e. in the constructor you say `self._msg_printer = MessagePrinter()` and then reference that instance variable in the methods which need it.